security-privacy.html

```diff
@@ -207,90 +211,161 @@
 <section class="informative"> <h2>Implementation details</h2>
   <p>
     This section contains discussion of implementation possibilities of security
-    related mechanisms.
+    related mechanisms. The following implementation alternatives are possible:
+    <ul>
+      <li>
+        Use additional NDEF records in an NDEF message, containing
+        metadata specific to Web NFC specific, such as identification, and
+        whitelists for reading and writing.
+      </li>
+        Use a specific Web NFC record format and encode the metadata specific to
+        Web NFC into that.
+      </li>
+    </ul>
   </p>
-  <section> <h3>Web NFC tags</h3>
+  <section> <h3>Web NFC using additional NDEF records for metadata</h3>
     <p>
-      For making tags that can be read and written by the Web NFC API, and for
-      making sure that web pages cannot access generic NFC tags, the following
-      are possible:
+      A typical Web NFC NDEF message will contain the following
+      NDEF records:
       <ul>
-        <li>Use the NDEF record <code>id</code> field in order to store the
-          origin of the page creating the record, which is also by default
-          allowed to modify the tag.</li>
-        <li>Use additional special NDEF records in each Web NFC tag, which are
-          hidden by the implementation. This solution depends on support for
-          writing multiple NDEF records per NDEF message.</li>
-        <li>Use additional data encoded into the NFC payload in a way which is
-          specific to Web NFC API. This solution is not desirable.</li>
+        <li>Normal NDEF records carrying data.</li>
+        <li>
+          An additional NDEF record for Web NFC identification.
+          The preferred type of this NDEF record is NFC Forum External Type
+          (TNF=0x04), with the External Type field set to
+          <code>urn:nfc:ext:w3.org:webnfc-id</code>.
+        </li>
+        <li>
+          Optionally, a NDEF record representing a read whitelist.
+          The preferred type of this NDEF record is NFC Forum External Type
+          (TNF=0x04), with the External Type field set to
+          <code>urn:nfc:ext:w3.org:webnfc-rwl</code>.
+          The payload is a string list of URL's, delimited by white space.
```

Comment:
Should these be URLs or [ascii-serialized Origins](https://html.spec.whatwg.org/multipage/browsers.html#ascii-serialisation-of-an-origin)?